test_cdk_pgstac_infra.py

```diff
@@ -0,0 +1,176 @@
+    "properties":{"datetime":"2023-02-15T00:00:00+00:00","proj:bbox":[4598521.999999994,5643304.000000009,4688521.999999994,5733304.000000009],"proj:shape":[3000,3000],"proj:geometry":{"type":"Polygon","coordinates":[[[4598521.999999994,5643304.000000009],[4688521.999999994,5643304.000000009],[4688521.999999994,5733304.000000009],[4598521.999999994,5733304.000000009],[4598521.999999994,5643304.000000009]]]},"proj:transform":[30.0,0.0,4598521.999999994,0.0,-30.0,5733304.000000009,0.0,0.0,1.0]},
+    "stac_extensions":["https://stac-extensions.github.io/projection/v1.1.0/schema.json","https://stac-extensions.github.io/raster/v1.1.0/schema.json"],
+    "geometry":{"type":"Polygon","coordinates":[[[14.114837413118664,67.218607039971],[12.438229073696998,67.70894310918132],[11.17860397724852,67.06631312684836],[12.837754767891637,66.58867761064732],[14.114837413118664,67.218607039971]]]}
+}
+
+
+#Validating the test collection
+try:
+    pystac.validation.validate_dict(test_collection)
+    print("Valid test collection")
+except:
+    raise STACValidationError
+
+#Validating the test item
+try:
+    pystac.validation.validate_dict(test_item)
+    print("Valid test item")
+except:
+    raise STACValidationError
+
+#Authentication - Get secret value
+stage = "dev"
+stac_register_service_id = "MAAP-workflows"
+stack_name = f"MAAP-STAC-auth-{stage}"
+
+# session = boto3.session.Session()
+client = boto3.client("secretsmanager", region_name='us-west-2')
+secret_id = f"{stack_name}/{stac_register_service_id}"
+
+try:
+    res_secret = client.get_secret_value(SecretId=secret_id)
+except client.exceptions.ResourceNotFoundException:
+    raise Exception(
+        f"Unable to find a secret for '{secret_id}'. "
+        "\n\nHint: Check your stage and service id. Also, verify that the correct "
+        "AWS_PROFILE is set on your environment."
+    )
+
+#Authentication - Get TOKEN
+secret = json.loads(res_secret["SecretString"])
+client_secret = secret["client_secret"]
+client_id = secret["client_id"]
+cognito_domain = "https://maap-stac-auth-dev.auth.us-west-2.amazoncognito.com"
+scope = "MAAP-STAC-ingestion-registry-server/stac:register"
+
+res_token = requests.post(
+            f"{cognito_domain}/oauth2/token",
+            headers={
+                "Content-Type":"application/x-www-form-urlencoded",
+            },
+            auth=(client_id, client_secret),
+            data={
+                "grant_type":"client_credentials",
+                # A space-separated list of scopes to request for the generated access token.
+                "scope": scope,
+            },
+        )
+
+TOKEN = res_token.json()["access_token"]
+
+#Insert the test collection
+session = requests.Session()
+base_url = "https://wynt4rero2.execute-api.us-west-2.amazonaws.com/test"
```

Comment:
Might want to change this to stac.dit.maap-project.org instead of the AWS API Gateway endpoint

Reply:
looks like this is resolved. Can you resolve discussions when you get a chance @omshinde ?